### ISSUE TITLE
Simple sequential "hyperopt" (sequential backtesting)

### ISSUE THREAD

**Issue by epigramx** [freqtrade/freqtrade#6510]
## Describe your environment (if applicable)
not applicable (it refers to all environments)

## Describe the enhancement

A way to get all combinations of backtesting from a certain linearly generated set of parameters; the current hyperopt code can be used or largely re-used for that; for example: you would set an Integer parameter between 1 and 10 and a step of 1 and after it is done: you can refer to all 10 backtesting results between 1 and 10 with `freqtrade hyperopt-show -n <1 to 10>` (or another command name or formulation if it must be renamed or reformulated).

The big advantage over current hyperopt is that it would be superior for the cases the user literally needs to see all results; it would also not drop certain combinations based on random seed; the big advantage over current backtesting: is that it doesn't have to be run manually multiple times.

If it has already been discussed here: point me to the #.

**Comment by xmatthias** [comment]
If there's an estimator for this (skopt or similar) - you should be able to use the [`generate_estimator`](https://www.freqtrade.io/en/stable/advanced-hyperopt/#overriding-base-estimator) override to define your own method of optimizing things.
As long as it implements `RegressorMixin` - you should also be able to supply "whatever" class/function - but details for this must be taken from the scikit-optimize documentation (which in all honesty - is unfortunately not that great).

I think [`dummy_minimize`](https://scikit-optimize.github.io/stable/modules/generated/skopt.dummy_minimize.html#skopt.dummy_minimize) might do something like this - but i've never tried that and didn't look at the implementation details.

There's also 3rd party modules which implement this IIRC and use different algorithms which are not available via scikit-learn/scikit-optimize directly.

While i'm not against adding more points/samples in this regards to the documentation if it provides certain benefits (like a proper grid search) - we will not add code for this to the freqtrade repository (implementing a regressor) which we will then have to maintain. 

If any - it'll have to be a separate project implementing the scikit-optimize interface and implementing whatever algorithm you imagine (but i assume such things should exist if you search "deep enough" in scikit-optimize issues).
The main reason for this is that such regressors are generally applicable (also outside of the freqtrade context).

**Comment by epigramx** [comment]
I tried a few settings of that estimator, and It appears to still repeat the same parameters occasionally, but I guess skopt might be beyond the scope of this project/ticket.

**Comment by xmatthias** [comment]
definitely beyond our scope, yes.

For someone familiar with scikit-optimize, it should be pretty simple to write a grid-search optimizer - simply combining all values with all results (if the dummy_minimize is not doing that yet).

You'll however always have "same parameters" occasionally - as at some point, you'll run out of points - and we (intentionally) don't do automatic early stopping - as with the regular algos, you can get the same points without having the space exhausted.

---

An alternative way might be a dummy loss function - which simply returns the same value all the time.

![image](https://user-images.githubusercontent.com/5024695/159152482-f611fbfd-6bff-4791-9aaf-39b8b4c2d222.png)


If you imagine the search-space geographically - as a map with hills and valleys - regular optimizers will try to minimize (speak - find the deepest valley) in this map.

If the loss function does return the same value all the time - then we'd create a flat map - where no minimum exists - so all points would have to be tried at random (obviously without ever finding a "best" result).

It should work in theory - but i guess it'll depend on the search-space size if you'll hit ALL points in a reasonable time - and how likely it is to retry combinations (probably not much - as it keeps a history of already attempted points).

**Comment by epigramx** [comment]
That hack appears to still allow it to repeat a parameter and the same happens with 'grid' or 'tree' settings in other estimators. It's weird to me that this is so common in those algorithms, because one would think they might have a memory to not repeat a parameter if it's only one parameter tested.

**Comment by xmatthias** [comment]
testing only one parameter with a very limited range is an edge-case - they're made to test potentially large spaces.
Scikit-optimize does also have no builtin "decimal" space - so you either have to use integers, or you're stuck with 16 decimal precision floats - which is a total waste of computing resources.

**Comment by epigramx** [comment]
I don't think this should use skopt at all. It's a simple looping through backtesting itself. For the time being I guess one could use the backtesting command line (but with the issue of not having the advantage of hyperopt having shared-memory for all tests).
